### PR TITLE
Enable Ruff rules - Part III

### PR DESCRIPTION
**Pull request by samet-akcay**
# Checklist

- [x] "UP",   # pyupgrade (`UP`)
- [x] "YTT",  # flake8-2020 (`YTT`)
- [x] "S",    # flake8-bandit (`S`)
- [x] "BLE",  # flake8-blind-except (`BLE`)
- [x] "FBT",  # flake8-boolean-trap (`FBT`)
- [x] "B",    # flake8-bugbear (`B`)

### CODE REVIEW

File: tools/inference/torch_inference.py [hunk 1]
```diff
@@ -85,7 +85,8 @@ def infer(args: Namespace) -> None:
 
         if args.output is None and args.show is False:
             warnings.warn(
```
Comment (ashwinvaidya17) [inline tools/inference/torch_inference.py:87]: Can we replace all warnings.warn to `logger.warn`? This way we will be able to redirect them to a file if needed and use rich's console logging to print the warnings.

File: src/anomalib/utils/callbacks/thresholding.py [hunk 2]
```diff
@@ -23,11 +23,9 @@ class _ThresholdCallback(Callback):
...
     def __init__(
         self,
-        threshold: BaseThreshold
-        | tuple[BaseThreshold, BaseThreshold]
-        | DictConfig
-        | ListConfig
-        | str = F1AdaptiveThreshold(),
+        threshold: BaseThreshold | tuple[BaseThreshold, BaseThreshold] | DictConfig | ListConfig
+        # TODO: Remove this noqa
+        | str = F1AdaptiveThreshold(),  # noqa: B008
```
Comment (ashwinvaidya17) [inline src/anomalib/utils/callbacks/thresholding.py:28]: How about we keep the default as a string `"F1AdaptiveThreshold"`

**Review by ashwinvaidya17 — COMMENTED** [review]
Wow a lot of changes

File: tools/train.py [hunk 3]
```diff
@@ -85,3 +85,4 @@ def train(args: Namespace):
 if __name__ == "__main__":
     args = get_parser().parse_args()
     train(args)
+    train(args)
```
Comment (ashwinvaidya17) [inline tools/train.py:88]: Might be extra?

File: src/anomalib/utils/metrics/__init__.py [hunk 4]
```diff
@@ -176,12 +180,13 @@ def create_metric_collection(
...
         assert all(isinstance(metric, str) for metric in metrics), f"All metrics must be strings, found {metrics}"
         return metric_collection_from_names(metrics, prefix)
 
-    if isinstance(metrics, (DictConfig, dict)):
+    if isinstance(metrics, DictConfig | dict):
         _validate_metrics_dict(metrics)
         return metric_collection_from_dicts(metrics, prefix)
 
     raise ValueError(f"metrics must be a list or a dict, found {type(metrics)}")
+    raise ValueError(f"metrics must be a list or a dict, found {type(metrics)}")
```
Comment (ashwinvaidya17) [inline src/anomalib/utils/metrics/__init__.py:192]: same

File: src/anomalib/data/utils/split.py [hunk 5]
```diff
@@ -132,3 +135,4 @@ def split_by_label(dataset: "data.AnomalibDataset") -> tuple["data.AnomalibDatas
     normal_subset = dataset.subsample(list(normal_indices))
     anomalous_subset = dataset.subsample(list(anomalous_indices))
     return normal_subset, anomalous_subset
+    return normal_subset, anomalous_subset
```
Comment (ashwinvaidya17) [inline src/anomalib/data/utils/split.py:138]: same

File: src/anomalib/data/utils/image.py [hunk 6]
```diff
@@ -243,3 +246,4 @@ def pad_nextpow2(batch: Tensor) -> Tensor:
     padding_h = [math.ceil((l_dim - batch.shape[-1]) / 2), math.floor((l_dim - batch.shape[-1]) / 2)]
     padded_batch = F.pad(batch, pad=[*padding_h, *padding_w])
     return padded_batch
+    return padded_batch
```
Comment (ashwinvaidya17) [inline src/anomalib/data/utils/image.py:249]: Maybe something is wrong in my diff

**Review by ashwinvaidya17 — COMMENTED** [review]
Thanks. Minor comments

File: tools/train.py (hunk 3 above)
Comment (samet-akcay) [reply tools/train.py:88]: weird, I haven't changed this file :/

Comment (samet-akcay) [reply tools/train.py]: anyway, fixed

**Review by ashwinvaidya17 — APPROVED** [review]

**Pull request merged**